tests/config/config_parser_test.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime
+import json
+import os
+from pathlib import Path
+import pytest
+from coast.config import ConfigParser
+from coast.config.config_structure import (
+    ConfigTypes,
+    ConfigKeys,
+    GriddedConfig,
+    IndexedConfig,
+    Dataset,
+    Domain,
+)
+
+# Valid gridded config json.
+gridded_json = {
+    "type": "gridded",
+    "grid_ref": {},
+    "dimensionality": 3,
+    "chunks": [],
+    "dataset": {"variable_map": {}, "dimension_map": {}},
+    "domain": {
+        "variable_map": {},
+        "dimension_map": {},
+    },
+    "processing_flags": [],
+}
+
+# Valid indexed config json.
+indexed_json = {
+    "type": "indexed",
+    "dimensionality": 3,
+    "chunks": [],
+    "dataset": {"variable_map": {}, "dimension_map": {}},
+    "processing_flags": [],
+}
+
+# Json with an invalid type value.
+invalid_type_json = {"type": "invalid"}
+
+
+@pytest.fixture
+def json_file(input_json):
+    """Write example json to file, for use with ConfigParser(). File auto deleted after test."""
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    cur_time = datetime.now().strftime("%Y%m%d%H%M%S")
+    tempfile = Path(f"{dir_path}/tempfile_{cur_time}.json")
+    with open(tempfile, "w") as temp:
```

Comment:
We can drop the with block and use `tempfile.write_text(json.dumps(input_json))`.

Reply:
Will notify Rob.